oauth2/provider/login.php

```diff
@@ -10,27 +10,13 @@
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  */
 
-require_once(__DIR__ . "/../../_rest_config.php");
-$gbl = RestConfig::GetInstance();
-
-$_GET['site'] = $gbl::$SITE;
-$ignoreAuth = true;
-require_once __DIR__ . '/../../interface/globals.php';
-
-use OpenEMR\Core\Header;
-
-$redirect = '';
-if ($_SESSION['get_credentials']) {
-    $redirect = $gbl::$REST_FULLAUTH_URL . "/login";
-    unset($_SESSION['get_credentials']);
-    $mode = 'credentials';
-} elseif ($_SESSION['get_authorization']) {
-    $redirect = $gbl::$REST_FULLAUTH_URL . "/device/code";
-    $authorize = 'authorize';
-} else {
+if ($oauthLogin !== true) {
```

Comment:
nice. don't think I spent much time on this and this is much cleaner. I wanted to leave some leadway because eventually this form will be shared from elsewhere besides our provider i.e a third party or from resource server for a reauth.

Reply:
yep, i just figured this was part of the long windy road that was required to get to your awesome solution (btw, I almost danced when I got a Bearer token :) ). This basically means the only way this script can be called is via the controller.